api/session_token_login_provider.go

```diff
@@ -73,11 +75,19 @@ func (p *sessionTokenLoginProvider) Login(ctx context.Context, caller base.APICa
 	return nil, errors.Trace(err)
 }
 
-func (p *sessionTokenLoginProvider) initiateDeviceLogin(ctx context.Context, caller base.APICaller) error {
-	if p.printOutputFunc == nil {
+func (p *sessionTokenLoginProvider) printOutput(format string, params ...any) error {
```

Comment:
`writeOutput` no printing is happening here.

Reply:
There is a `Fprint` a few lines down.